pkg/controller/blockdevice/effects.go

```diff
@@ -0,0 +1,423 @@
+			diskToRemove, ok := node.Spec.Disks[bd.Name]
+			if !ok {
+				return onUnprovisionFinished(bd)
+			}
+
+			isUnprovisioning := false
+			for _, tag := range diskToRemove.Tags {
+				if tag == util.DiskRemoveTag {
+					isUnprovisioning = true
+					break
+				}
+			}
+
+			if !isUnprovisioning {
+				logEffect(bd).Info("Start unprovisioning")
+				diskToRemove.AllowScheduling = false
+				diskToRemove.EvictionRequested = true
+				diskToRemove.Tags = append(diskToRemove.Tags, util.DiskRemoveTag)
+				nodeCpy := node.DeepCopy()
+				nodeCpy.Spec.Disks[bd.Name] = diskToRemove
+				newNode, err := e.Nodes().Update(nodeCpy)
+				if err != nil {
+					// If a critical resource is missing, then abort this polling
+					// because we cannot update its state anymore
+					return errors.IsNotFound(err), err
+				}
+				node = newNode
+				return false, nil
+			}
+
+			if status, ok := node.Status.DiskStatus[bd.Name]; ok && len(status.ScheduledReplica) > 0 {
+				logEffect(bd).Info("Still unprovisioning")
+				return false, nil
+			}
+
+			nodeCpy := node.DeepCopy()
+			delete(nodeCpy.Spec.Disks, bd.Name)
+			if !reflect.DeepEqual(nodeCpy.Spec.Disks, node.Spec.Disks) {
+				if _, err := e.Nodes().Update(nodeCpy); err != nil {
+					if errors.IsConflict(err) {
+						// Hit optimistic lock. Fetch latest resource for next poll.
+						newNode, err := e.NodeCache().Get(node.Namespace, node.Name)
+						if err != nil {
+							// If a critical resource is missing, then abort this polling
+							// because we cannot update its state anymore
+							return errors.IsNotFound(err), err
+						}
+						node = newNode
+					}
+					return false, err
+				}
+			}
+			// Finish. The device should back to mounted.
+			return onUnprovisionFinished(bd)
+		})
+		return nil
+	}
+}
+
+func getNode(e effectController, bd *diskv1.BlockDevice) (*longhornv1.Node, error) {
+	node, err := e.NodeCache().Get(bd.Namespace, bd.Spec.NodeName)
+	if err != nil && errors.IsNotFound(err) {
+		node, err = e.Nodes().Get(bd.Namespace, bd.Spec.NodeName, metav1.GetOptions{})
+	}
+	return node, err
+}
+
+func logEffect(bd *diskv1.BlockDevice) *logrus.Entry {
+	return logrus.WithFields(logrus.Fields{
+		"device":  bd.Name,
+		"context": "Effect",
+	})
+}
+
+// jitterPoll polls the conditionFunc with time jittering.
+// The conditionFunc here has a slight different semantic: if `done` is true,
+// no matter there is an error with it or not, jitterPoll always ceases.
+func jitterPoll(e effectController, bd *diskv1.BlockDevice, conditionFunc wait.ConditionFunc) {
+	stopCh := make(chan struct{})
+	doneCh := make(chan struct{})
+	currentPhase := bd.Status.ProvisionPhase
+	go func() {
+		wait.JitterUntil(func() {
+			done, err := conditionFunc()
+			if done {
+				if err != nil {
+					logEffect(bd).Errorf("Ceased phase %s with err: %v", currentPhase, err.Error())
+				} else {
+					logEffect(bd).Infof("Finished phase %s", currentPhase)
+				}
+				stopCh <- struct{}{}
```

Comment:
The document suggest closing the stopCh to stop JitterUntil [REF](https://github.com/kubernetes/apimachinery/blob/4427f8f31dfbac65d3a044d0168f84c51bfda440/pkg/util/wait/wait.go#L130)
```suggestion
				close(stopCh)
```